inc/hclib-promise.h

```diff
@@ -73,31 +73,15 @@ typedef struct _hclib_future_t {
     struct hclib_promise_st *owner;
 } hclib_future_t;
 
-/**
- * An hclib_triggered_task_t associates dependent tasks and the promises that
- * trigger their execution. This is exposed so that the runtime knows the size
- * of the struct.
- */
-typedef struct hclib_triggered_task_st {
-    // NULL-terminated list of futures this task is registered on
-    hclib_future_t **waiting_frontier;
-    /*
-     * This allows us to chain all dependent tasks waiting on a same promise.
-     * Whenever a triggered task wants to register on a promise, and that
-     * promise is not ready, we chain the current triggered task and the
-     * promise's wait_list_head and try to cas on the promise's wait_list_head,
-     * with the current triggered task.
-     */
-    struct hclib_triggered_task_st *next_waiting_on_same_future;
-} hclib_triggered_task_t;
+struct hclib_task_t;
 
 // We define a typedef in this unit for convenience
 typedef struct hclib_promise_st {
     hclib_future_t future;
-	int kind;
-    volatile void *datum;
+    promise_kind_t kind;
+    void *volatile datum;
```

Comment:
Does it make sense for the pointer itself to be volatile? It seems like its value will only ever be set once and never changed again, what's the case where having the pointer storage itself be volatile is necessary?